### DISCUSSION TITLE
Unable to add sorting ability to table

### DISCUSSION THREAD

**Discussion by bartclaeys** [filamentphp/filament#5089] [General]
I am trying to make my table sortable by following the following instructions:

https://filamentphp.com/tricks/sorting-table-with-spatie-eloquent-sortable
https://filamentphp.com/docs/2.x/admin/resources/listing-records#reordering-records
https://github.com/spatie/eloquent-sortable

But, it's not working. Questions:

1. How is this supposed to look like? Doc says: "a new button will be available on the table to toggle reordering" but unclear where that button would appear and how the interaction would work. A screenshot or video would be helpful.

2. Should the button appear without even using Spatie's Sortable package being installed?

3. Does Spatie's grouping work within Filament tables?

**Comment by RX-77IT** [comment]
Hi, I can answer only to the first two questions.
1. The button appears in the top left corner of the table.
![order button](https://i.imgur.com/sNlo2H3.png)
When you press it, the records in the table become orderable.
![order button pressed](https://i.imgur.com/ASaVEsq.png)

2. I'm using the base Filament package.

**Comment by danharrin** [comment]
What is Spatie's grouping?

**Reply by bartclaeys** [reply]
Let's say you have a table of books, you probably don't want to order these all together, but order in their respective groups. These books eg. belong to a category (eg. Fiction) in which you want to reorder the books.

Spatie's package allows to define a foreign key to group by. But it's unclear how that works together with Filament.

**Reply by danharrin** [reply]
no sorry, that's not possible at the moment. you would instead reorder the categories in the CategoryResource

**Reply by bartclaeys** [reply]
Note sure whether I explained it well. It's about the order of the books within a category. Not reordering the categories themselves.

**Reply by danharrin** [reply]
For that, I would suggest adding a category filter. Filter by a category, and then reorder the books :) That would work, right?

**Comment by wivaku** [comment]
For me it is not working either. I do see the indicator, and I can drag a row.  
But when I let go the row it returns back to the original location.

I am using base Filament (v2.16.50) -without Spatie-, and a model that has a column `order_column`.
In my resource page for that model I have:
```
    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                // 
            ])
            // ->defaultSort('order_column', 'asc') // not needed, using global scope
            ->reorderable('order_column')
            ->filters([
                // 
            ]);
    }
```
Am I missing something?

**Reply by RX-77IT** [reply]
This happened to me once: turns out i had forgotten to put `order_column` in the `$fillable` property of the model. XD

**Reply by wivaku** [reply]
Of course! That was it.
